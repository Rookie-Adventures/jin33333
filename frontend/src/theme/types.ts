import type { Theme as MuiTheme, ThemeOptions } from '@mui/material/styles';
import type { TypographyStyle } from '@mui/material/styles/createTypography';
import type { CSSObject } from '@mui/styled-engine';
import type { Theme as SystemTheme } from '@mui/system/createTheme';

export type ThemeMode = 'light' | 'dark';

// 扩展 Typography 选项
export interface ExtendedTypographyOptions {
  fontFamily: string;
  h1: TypographyStyle;
  h2: TypographyStyle;
  h3: TypographyStyle;
  h4: TypographyStyle;
  h5: TypographyStyle;
  h6: TypographyStyle;
  body1: TypographyStyle;
  body2: TypographyStyle;
  button: TypographyStyle;
}

// 扩展基础主题类型
export interface BaseTheme extends Omit<MuiTheme, 'typography'> {
  typography: ExtendedTypographyOptions;
}

// 扩展主题选项类型
export interface CustomThemeOptions extends Omit<ThemeOptions, 'typography'> {
  typography?: Partial<ExtendedTypographyOptions>;
}

// 扩展完整主题类型
export type CustomTheme = BaseTheme & {
  components?: {
    MuiButton?: {
      styleOverrides?: {
        root?: Partial<CSSObject>;
      };
    };
    MuiTextField?: {
      styleOverrides?: {
        root?: Partial<CSSObject>;
      };
    };
    MuiCard?: {
      styleOverrides?: {
        root?: Partial<CSSObject>;
      };
    };
    [key: string]: {
      styleOverrides?: {
        root?: Partial<CSSObject>;
      };
    } | undefined;
  };
};

// 使用 MUI System 的主题类型
export type Theme = SystemTheme;
