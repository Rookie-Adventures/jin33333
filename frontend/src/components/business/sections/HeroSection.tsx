import { AutoAwesome, Speed, CloudDone } from '@mui/icons-material';
import { Box, Typography, Button, Grid, useTheme, useMediaQuery, Container } from '@mui/material';
import { motion } from 'framer-motion';
import React from 'react';

const HeroSection: React.FC = () => {
  const theme = useTheme();
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));

  return (
    <Container maxWidth="lg">
      <Box
        sx={{
          width: '100%',
          py: { xs: 4, md: 8 },
          px: { xs: 2, sm: 3 },
        }}
      >
        <Grid container spacing={4} alignItems="center">
          <Grid item xs={12} md={6}>
            <Box
              component={motion.div}
              initial={{ opacity: 0, x: -50 }}
              animate={{ opacity: 1, x: 0 }}
              transition={{ duration: 0.5 }}
            >
              <Typography
                variant={isMobile ? 'h3' : 'h2'}
                component="h1"
                gutterBottom
                sx={{ fontWeight: 'bold' }}
              >
                优质的 Jinsansan 模型调用体验
              </Typography>
              <Typography variant="h5" color="inherit" paragraph sx={{ opacity: 0.9, mb: 4 }}>
                为中国用户提供稳定、快速的 AI 模型服务，支持流式响应，智能上下文管理
              </Typography>
              <Box sx={{ display: 'flex', gap: 3, mb: 4 }}>
                <Box sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
                  <Speed />
                  <Typography>快速响应</Typography>
                </Box>
                <Box sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
                  <CloudDone />
                  <Typography>稳定可靠</Typography>
                </Box>
                <Box sx={{ display: 'flex', alignItems: 'center', gap: 1 }}>
                  <AutoAwesome />
                  <Typography>智能管理</Typography>
                </Box>
              </Box>
              <Button
                variant="contained"
                color="secondary"
                size="large"
                sx={{
                  borderRadius: 2,
                  px: 4,
                  py: 1.5,
                  fontSize: '1.1rem',
                  textTransform: 'none',
                }}
              >
                立即体验
              </Button>
            </Box>
          </Grid>
          <Grid item xs={12} md={6}>
            <Box
              component={motion.div}
              initial={{ opacity: 0, x: 50 }}
              animate={{ opacity: 1, x: 0 }}
              transition={{ duration: 0.5, delay: 0.2 }}
              sx={{
                display: 'flex',
                justifyContent: 'center',
                alignItems: 'center',
                height: '100%',
              }}
            >
              <Box
                component="img"
                src="/ai-chat.svg"
                alt="AI Chat Illustration"
                sx={{
                  width: '100%',
                  maxWidth: 500,
                  height: 'auto',
                  filter: 'drop-shadow(0px 4px 20px rgba(0, 0, 0, 0.1))',
                }}
              />
            </Box>
          </Grid>
        </Grid>
      </Box>
    </Container>
  );
};

export default HeroSection;
