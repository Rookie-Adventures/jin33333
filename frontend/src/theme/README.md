# 主题模块 (Theme)

## 目录结构

```
theme/
├── index.ts        # 主题配置入口
├── types.ts        # 主题类型定义
├── utils.ts        # 主题工具函数
├── index.test.ts   # 主题配置测试
└── utils.test.ts   # 工具函数测试
```

## 主题规范

### 命名规范
- 主题文件使用 camelCase 命名（如 `themeConfig.ts`）
- 类型文件使用 camelCase 命名（如 `themeTypes.ts`）
- 工具文件使用 camelCase 命名（如 `themeUtils.ts`）
- 测试文件使用 `.test.ts` 后缀

### 主题结构
```typescript
// 主题配置模板
import { createTheme } from '@mui/material/styles';

export const theme = createTheme({
  palette: {
    primary: {
      main: '#1976d2',
      light: '#42a5f5',
      dark: '#1565c0',
    },
    secondary: {
      main: '#9c27b0',
      light: '#ba68c8',
      dark: '#7b1fa2',
    },
  },
  typography: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    h1: {
      fontSize: '2.5rem',
      fontWeight: 500,
    },
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          borderRadius: 8,
        },
      },
    },
  },
});
```

## 主题分类

### 1. 主题配置 (index.ts)
- 颜色系统
- 排版系统
- 组件样式
- 响应式断点

### 2. 类型定义 (types.ts)
- 主题类型
- 组件类型
- 工具类型

### 3. 工具函数 (utils.ts)
- 颜色工具
- 排版工具
- 响应式工具

## 使用示例

### 主题配置
```typescript
import { ThemeProvider } from '@mui/material/styles';
import { theme } from './theme';

export const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <CssBaseline />
      <AppContent />
    </ThemeProvider>
  );
};
```

### 主题使用
```typescript
import { useTheme } from '@mui/material/styles';

export const ThemedComponent = () => {
  const theme = useTheme();
  
  return (
    <Box sx={{
      color: theme.palette.primary.main,
      typography: theme.typography.h1,
    }}>
      Themed Content
    </Box>
  );
};
```

### 响应式使用
```typescript
import { useMediaQuery } from '@mui/material';

export const ResponsiveComponent = () => {
  const isMobile = useMediaQuery(theme.breakpoints.down('sm'));
  
  return (
    <Box sx={{
      padding: isMobile ? 2 : 4,
    }}>
      Responsive Content
    </Box>
  );
};
```

## 最佳实践

1. **主题设计**
   - 使用设计系统
   - 保持一致性
   - 支持暗色模式

2. **性能优化**
   - 优化主题切换
   - 减少样式计算
   - 使用 CSS-in-JS

3. **测试规范**
   - 编写单元测试
   - 测试主题切换
   - 测试响应式

4. **开发规范**
   - 使用 TypeScript
   - 遵循 Material-UI 最佳实践
   - 保持代码简洁

## 注意事项

1. 主题开发时需要考虑：
   - 主题的可维护性
   - 主题的性能表现
   - 主题的类型安全
   - 主题的测试覆盖

2. 主题使用时的注意事项：
   - 正确处理主题切换
   - 处理响应式布局
   - 遵循主题规范

## 更新日志

### v1.0.0
- 初始化主题模块
- 实现基础主题
- 添加主题文档 