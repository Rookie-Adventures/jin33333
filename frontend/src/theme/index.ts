import { createTheme as createMuiTheme } from '@mui/material/styles';
import type { ThemeOptions } from '@mui/material/styles';

export const defaultTheme: ThemeOptions = {
  palette: {
    mode: 'light' as const,
    primary: {
      main: '#1976d2',
      light: '#42a5f5',
      dark: '#1565c0',
      contrastText: '#ffffff'
    },
    secondary: {
      main: '#9c27b0',
      light: '#ba68c8',
      dark: '#7b1fa2',
      contrastText: '#ffffff'
    },
    error: {
      main: '#d32f2f',
      light: '#ef5350',
      dark: '#c62828',
      contrastText: '#ffffff'
    },
    warning: {
      main: '#ed6c02',
      light: '#ff9800',
      dark: '#e65100',
      contrastText: '#ffffff'
    },
    info: {
      main: '#0288d1',
      light: '#03a9f4',
      dark: '#01579b',
      contrastText: '#ffffff'
    },
    success: {
      main: '#2e7d32',
      light: '#4caf50',
      dark: '#1b5e20',
      contrastText: '#ffffff'
    },
    background: {
      default: '#ffffff',
      paper: '#f5f5f5'
    },
    text: {
      primary: 'rgba(0, 0, 0, 0.87)',
      secondary: 'rgba(0, 0, 0, 0.6)'
    }
  },
  typography: {
    fontFamily: '"Roboto", "Helvetica", "Arial", sans-serif',
    h1: {
      fontSize: '2.5rem',
      fontWeight: 500
    },
    h2: {
      fontSize: '2rem',
      fontWeight: 500
    },
    h3: {
      fontSize: '1.75rem',
      fontWeight: 500
    },
    h4: {
      fontSize: '1.5rem',
      fontWeight: 500
    },
    h5: {
      fontSize: '1.25rem',
      fontWeight: 500
    },
    h6: {
      fontSize: '1rem',
      fontWeight: 500
    },
    body1: {
      fontSize: '1rem',
      lineHeight: 1.5
    },
    body2: {
      fontSize: '0.875rem',
      lineHeight: 1.43
    },
    button: {
      fontSize: '0.875rem',
      textTransform: 'uppercase',
      fontWeight: 500
    }
  },
  spacing: 8,
  breakpoints: {
    values: {
      xs: 0,
      sm: 600,
      md: 960,
      lg: 1280,
      xl: 1920
    }
  },
  transitions: {
    easing: {
      easeInOut: 'cubic-bezier(0.4, 0, 0.2, 1)',
      easeOut: 'cubic-bezier(0.0, 0, 0.2, 1)',
      easeIn: 'cubic-bezier(0.4, 0, 1, 1)',
      sharp: 'cubic-bezier(0.4, 0, 0.6, 1)'
    },
    duration: {
      shortest: 150,
      shorter: 200,
      short: 250,
      standard: 300,
      complex: 375,
      enteringScreen: 225,
      leavingScreen: 195
    }
  },
  components: {
    MuiButton: {
      styleOverrides: {
        root: {
          textTransform: 'none'
        } as const
      }
    },
    MuiTextField: {
      defaultProps: {
        variant: 'outlined'
      }
    },
    MuiCard: {
      styleOverrides: {
        root: {
          borderRadius: 8
        }
      }
    }
  }
};

export const darkTheme: ThemeOptions = {
  ...defaultTheme,
  palette: {
    ...defaultTheme.palette,
    mode: 'dark' as const,
    background: {
      default: '#121212',
      paper: '#1e1e1e'
    },
    text: {
      primary: '#ffffff',
      secondary: 'rgba(255, 255, 255, 0.7)'
    },
    action: {
      active: '#fff',
      hover: 'rgba(255, 255, 255, 0.08)',
      selected: 'rgba(255, 255, 255, 0.16)',
      disabled: 'rgba(255, 255, 255, 0.3)',
      disabledBackground: 'rgba(255, 255, 255, 0.12)'
    }
  }
};

export const createTheme = (isDark: boolean, customOptions: Partial<ThemeOptions> = {}): ThemeOptions => {
  const baseTheme = isDark ? darkTheme : defaultTheme;
  return createMuiTheme(baseTheme, customOptions);
};

// 导出默认主题实例
export const theme = createTheme(false); 