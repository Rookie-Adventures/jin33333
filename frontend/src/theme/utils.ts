import { Theme } from '@mui/material';
import { darken, lighten } from '@mui/material/styles';

/**
 * 主题切换动画持续时间（毫秒）
 */
export const THEME_TRANSITION_DURATION = 200;

/**
 * 获取主题过渡样式
 * @param theme - MUI主题对象
 * @returns 过渡样式对象
 */
export const getThemeTransitionStyles = (theme: Theme) => ({
  // 全局过渡效果
  '*': {
    transition: theme.transitions.create(
      ['background-color', 'color', 'border-color', 'box-shadow'],
      {
        duration: THEME_TRANSITION_DURATION,
        easing: theme.transitions.easing.easeInOut,
      }
    ),
  },
  // 特定元素过渡效果
  '.MuiPaper-root': {
    transition: theme.transitions.create(['background-color', 'box-shadow'], {
      duration: THEME_TRANSITION_DURATION,
      easing: theme.transitions.easing.easeInOut,
    }),
  },
  '.MuiAppBar-root': {
    transition: theme.transitions.create(['background-color', 'box-shadow'], {
      duration: THEME_TRANSITION_DURATION,
      easing: theme.transitions.easing.easeInOut,
    }),
  },
});

export const getContrastText = (background: string): string => {
  if (!isValidColor(background)) {
    throw new Error('Invalid color format');
  }
  const rgb = hexToRgb(background);
  const brightness = (rgb.r * 299 + rgb.g * 587 + rgb.b * 114) / 1000;
  return brightness >= 128 ? '#000000' : '#ffffff';
};

export const adjustColor = (color: string, amount: number): string => {
  if (amount < -1 || amount > 1) {
    throw new Error('Adjustment amount must be between -1 and 1');
  }
  const rgb = hexToRgb(color);
  const adjusted = {
    r: Math.min(
      255,
      Math.max(0, Math.round(rgb.r + (amount < 0 ? -rgb.r : 255 - rgb.r) * Math.abs(amount)))
    ),
    g: Math.min(
      255,
      Math.max(0, Math.round(rgb.g + (amount < 0 ? -rgb.g : 255 - rgb.g) * Math.abs(amount)))
    ),
    b: Math.min(
      255,
      Math.max(0, Math.round(rgb.b + (amount < 0 ? -rgb.b : 255 - rgb.b) * Math.abs(amount)))
    ),
  };
  return rgbToHex(adjusted);
};

export const createTransitions = (
  properties: string[],
  duration: number = 200,
  easing: string = 'ease'
): Record<string, string> => {
  return properties.reduce(
    (acc, prop) => ({
      ...acc,
      [prop]: `${duration}ms${easing ? ` ${easing}` : ''}`,
    }),
    {}
  );
};

export interface ColorPalette {
  main: string;
  light: string;
  dark: string;
  contrastText: string;
  alpha?: (opacity: number) => string;
}

export interface ExtendedColorPalette extends ColorPalette {
  secondary: ColorPalette;
}

export const generatePalette = (
  primaryColor: string,
  withSecondary: boolean = false
): ColorPalette | ExtendedColorPalette => {
  if (!isValidColor(primaryColor)) {
    throw new Error('Invalid color format');
  }

  const createAlphaFn = (color: string) => (opacity: number) => {
    const rgb = hexToRgb(color);
    return `rgba(${rgb.r}, ${rgb.g}, ${rgb.b}, ${Math.max(0, Math.min(1, opacity))})`;
  };

  const palette: ColorPalette = {
    main: primaryColor,
    light: lighten(primaryColor, 0.2),
    dark: darken(primaryColor, 0.2),
    contrastText: getContrastText(primaryColor),
    alpha: createAlphaFn(primaryColor),
  };

  if (withSecondary) {
    const secondaryColor = generateSecondaryColor(primaryColor);
    return {
      ...palette,
      secondary: {
        main: secondaryColor,
        light: lighten(secondaryColor, 0.2),
        dark: darken(secondaryColor, 0.2),
        contrastText: getContrastText(secondaryColor),
        alpha: createAlphaFn(secondaryColor),
      },
    };
  }

  return palette;
};

// 辅助函数
const hexToRgb = (hex: string) => {
  const shorthandRegex = /^#?([a-f\d])([a-f\d])([a-f\d])$/i;
  hex = hex.replace(shorthandRegex, (_, r, g, b) => r + r + g + g + b + b);

  const result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex);
  return result
    ? {
        r: parseInt(result[1], 16),
        g: parseInt(result[2], 16),
        b: parseInt(result[3], 16),
      }
    : { r: 0, g: 0, b: 0 };
};

const isValidColor = (color: string): boolean => {
  const hexRegex = /^#([A-Fa-f0-9]{6}|[A-Fa-f0-9]{3})$/;
  return hexRegex.test(color);
};

const generateSecondaryColor = (primaryColor: string): string => {
  // 简单的互补色生成
  const rgb = hexToRgb(primaryColor);
  const r = (255 - rgb.r).toString(16).padStart(2, '0');
  const g = (255 - rgb.g).toString(16).padStart(2, '0');
  const b = (255 - rgb.b).toString(16).padStart(2, '0');
  return `#${r}${g}${b}`;
};

const rgbToHex = (rgb: { r: number; g: number; b: number }): string => {
  const toHex = (n: number) => {
    const hex = Math.min(255, Math.max(0, n)).toString(16);
    return hex.length === 1 ? '0' + hex : hex;
  };
  return `#${toHex(rgb.r)}${toHex(rgb.g)}${toHex(rgb.b)}`;
};
