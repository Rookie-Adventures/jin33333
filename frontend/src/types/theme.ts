export type ThemeMode = 'light' | 'dark';

export interface ThemeCustomizations {
  primaryColor: string;
}

export interface ThemeState {
  mode: ThemeMode;
  customizations: ThemeCustomizations;
} 