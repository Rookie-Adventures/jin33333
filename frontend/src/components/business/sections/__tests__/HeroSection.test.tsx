import { render, screen } from '@testing-library/react';
import { ThemeProvider } from '@mui/material/styles';
import { useMediaQuery } from '@mui/material';
import { vi } from 'vitest';

import HeroSection from '../HeroSection';
import { createTheme } from '@/theme';

// Mock useMediaQuery
vi.mock('@mui/material', async () => {
  const actual = await vi.importActual<typeof import('@mui/material')>('@mui/material');
  return {
    ...actual,
    useMediaQuery: vi.fn(),
  };
});

describe('HeroSection', () => {
  const renderHeroSection = () => {
    return render(
      <ThemeProvider theme={createTheme(false)}>
        <HeroSection />
      </ThemeProvider>
    );
  };

  beforeEach(() => {
    // 默认设置为桌面视图
    vi.mocked(useMediaQuery).mockReturnValue(false);
  });

  it('应该正确渲染标题和副标题', () => {
    renderHeroSection();

    expect(screen.getByText(/优质的 Jinsansan 模型调用体验/i)).toBeInTheDocument();
    expect(screen.getByText(/为中国用户提供稳定、快速的 AI 模型服务/i)).toBeInTheDocument();
  });

  it('应该正确渲染特性标签', () => {
    renderHeroSection();

    expect(screen.getByText('快速响应')).toBeInTheDocument();
    expect(screen.getByText('稳定可靠')).toBeInTheDocument();
    expect(screen.getByText('智能管理')).toBeInTheDocument();
  });

  it('应该正确渲染 CTA 按钮', () => {
    renderHeroSection();

    const ctaButton = screen.getByText('立即体验');
    expect(ctaButton).toBeInTheDocument();
    expect(ctaButton).toHaveStyle({
      textTransform: 'none',
    });
  });

  it('应该在移动端正确响应', () => {
    // 模拟移动端视图
    vi.mocked(useMediaQuery).mockReturnValue(true);

    renderHeroSection();

    const title = screen.getByText(/优质的 Jinsansan 模型调用体验/i);
    expect(title).toHaveStyle({
      fontSize: expect.stringMatching(/h3/),
    });
  });

  it('应该正确渲染图片', () => {
    renderHeroSection();

    const image = screen.getByAltText('AI Chat Illustration');
    expect(image).toBeInTheDocument();
    expect(image).toHaveAttribute('src', '/ai-chat.svg');
  });

  it('应该正确应用动画效果', async () => {
    renderHeroSection();

    const contentContainer = screen.getByText(/优质的 Jinsansan 模型调用体验/i).parentElement;
    expect(contentContainer).not.toBeNull();

    // 等待动画完成
    await new Promise(resolve => setTimeout(resolve, 1500));

    const styles = window.getComputedStyle(contentContainer!);
    expect(styles.opacity).not.toBe('0');
    expect(styles.transform).not.toContain('translateX(-20px)');
  });

  // 添加快照测试
  it('应该匹配桌面端快照', () => {
    const { container } = renderHeroSection();
    expect(container).toMatchSnapshot();
  });

  it('应该匹配移动端快照', () => {
    vi.mocked(useMediaQuery).mockReturnValue(true);
    const { container } = renderHeroSection();
    expect(container).toMatchSnapshot();
  });
});
