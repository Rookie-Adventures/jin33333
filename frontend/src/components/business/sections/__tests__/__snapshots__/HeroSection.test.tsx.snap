// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HeroSection > 应该匹配桌面端快照 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-11b9611"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-xtlx1y-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
            style="opacity: 0; transform: translateX(-50px);"
          >
            <h1
              class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom css-197vo3u-MuiTypography-root"
            >
              优质的 Jinsansan 模型调用体验
            </h1>
            <p
              class="MuiTypography-root MuiTypography-h5 MuiTypography-paragraph css-i01w30-MuiTypography-root"
            >
              为中国用户提供稳定、快速的 AI 模型服务，支持流式响应，智能上下文管理
            </p>
            <div
              class="MuiBox-root css-9ct5jt"
            >
              <div
                class="MuiBox-root css-axw7ok"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SpeedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m20.38 8.57-1.23 1.85a8 8 0 0 1-.22 7.58H5.07A8 8 0 0 1 15.58 6.85l1.85-1.23A10 10 0 0 0 3.35 19a2 2 0 0 0 1.72 1h13.85a2 2 0 0 0 1.74-1 10 10 0 0 0-.27-10.44zm-9.79 6.84a2 2 0 0 0 2.83 0l5.66-8.49-8.49 5.66a2 2 0 0 0 0 2.83"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10utshc-MuiTypography-root"
                >
                  快速响应
                </p>
              </div>
              <div
                class="MuiBox-root css-axw7ok"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloudDoneIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M10 17l-3.5-3.5 1.41-1.41L10 14.17 15.18 9l1.41 1.41z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10utshc-MuiTypography-root"
                >
                  稳定可靠
                </p>
              </div>
              <div
                class="MuiBox-root css-axw7ok"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AutoAwesomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m19 9 1.25-2.75L23 5l-2.75-1.25L19 1l-1.25 2.75L15 5l2.75 1.25zm-7.5.5L9 4 6.5 9.5 1 12l5.5 2.5L9 20l2.5-5.5L17 12zM19 15l-1.25 2.75L15 19l2.75 1.25L19 23l1.25-2.75L23 19l-2.75-1.25z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10utshc-MuiTypography-root"
                >
                  智能管理
                </p>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSecondary css-3vsl00-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              立即体验
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-16r1tk9"
            style="opacity: 0; transform: translateX(50px);"
          >
            <img
              alt="AI Chat Illustration"
              class="MuiBox-root css-ex67qq"
              src="/ai-chat.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeroSection > 应该匹配移动端快照 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
  >
    <div
      class="MuiBox-root css-11b9611"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-4 css-xtlx1y-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
            style="opacity: 0; transform: translateX(-50px);"
          >
            <h1
              class="MuiTypography-root MuiTypography-h3 MuiTypography-gutterBottom css-4yr12r-MuiTypography-root"
            >
              优质的 Jinsansan 模型调用体验
            </h1>
            <p
              class="MuiTypography-root MuiTypography-h5 MuiTypography-paragraph css-i01w30-MuiTypography-root"
            >
              为中国用户提供稳定、快速的 AI 模型服务，支持流式响应，智能上下文管理
            </p>
            <div
              class="MuiBox-root css-9ct5jt"
            >
              <div
                class="MuiBox-root css-axw7ok"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SpeedIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m20.38 8.57-1.23 1.85a8 8 0 0 1-.22 7.58H5.07A8 8 0 0 1 15.58 6.85l1.85-1.23A10 10 0 0 0 3.35 19a2 2 0 0 0 1.72 1h13.85a2 2 0 0 0 1.74-1 10 10 0 0 0-.27-10.44zm-9.79 6.84a2 2 0 0 0 2.83 0l5.66-8.49-8.49 5.66a2 2 0 0 0 0 2.83"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10utshc-MuiTypography-root"
                >
                  快速响应
                </p>
              </div>
              <div
                class="MuiBox-root css-axw7ok"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CloudDoneIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19.35 10.04C18.67 6.59 15.64 4 12 4 9.11 4 6.6 5.64 5.35 8.04 2.34 8.36 0 10.91 0 14c0 3.31 2.69 6 6 6h13c2.76 0 5-2.24 5-5 0-2.64-2.05-4.78-4.65-4.96M10 17l-3.5-3.5 1.41-1.41L10 14.17 15.18 9l1.41 1.41z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10utshc-MuiTypography-root"
                >
                  稳定可靠
                </p>
              </div>
              <div
                class="MuiBox-root css-axw7ok"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="AutoAwesomeIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="m19 9 1.25-2.75L23 5l-2.75-1.25L19 1l-1.25 2.75L15 5l2.75 1.25zm-7.5.5L9 4 6.5 9.5 1 12l5.5 2.5L9 20l2.5-5.5L17 12zM19 15l-1.25 2.75L15 19l2.75 1.25L19 23l1.25-2.75L23 19l-2.75-1.25z"
                  />
                </svg>
                <p
                  class="MuiTypography-root MuiTypography-body1 css-10utshc-MuiTypography-root"
                >
                  智能管理
                </p>
              </div>
            </div>
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSecondary MuiButton-root MuiButton-contained MuiButton-containedSecondary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-colorSecondary css-3vsl00-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="button"
            >
              立即体验
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-16r1tk9"
            style="opacity: 0; transform: translateX(50px);"
          >
            <img
              alt="AI Chat Illustration"
              class="MuiBox-root css-ex67qq"
              src="/ai-chat.svg"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
